conbench/entities/history.py

```diff
@@ -85,31 +89,170 @@ def get_history(case_id, context_id, hardware_hash, repo) -> List[tuple]:
             BenchmarkResult.unit,
             BenchmarkResult.data,
             BenchmarkResult.times,
-            BenchmarkResult.change_annotations,
+            # turn Nones into Falses for this column
+            func.coalesce(
+                BenchmarkResult.change_annotations[
+                    "begins_distribution_change"
+                ].as_boolean(),
+                False,
+            ).label("begins_distribution_change"),
             Hardware.hash,
             Commit.sha,
             Commit.repository,
             Commit.message,
             Commit.timestamp,
-            Distribution.mean_mean,
-            Distribution.mean_sd,
             Run.name,
         )
         .join(Run, Run.id == BenchmarkResult.run_id)
         .join(Hardware, Hardware.id == Run.hardware_id)
         .join(Commit, Commit.id == Run.commit_id)
-        .join(Distribution, Distribution.commit_id == Commit.id)
         .filter(
             BenchmarkResult.case_id == case_id,
             BenchmarkResult.context_id == context_id,
             BenchmarkResult.error.is_(None),
             Commit.sha == Commit.fork_point_sha,  # on default branch
             Commit.repository == repo,
             Hardware.hash == hardware_hash,
-            Distribution.case_id == case_id,
-            Distribution.context_id == context_id,
-            Distribution.hardware_hash == hardware_hash,
         )
-        .order_by(Commit.timestamp.asc())
-        .all()
+        .subquery()
     )
+
+    # Add column with cumulative sum of distribution changes
+    history = Session.query(
+        history,
+        func.sum(history.c.begins_distribution_change.cast(s.Integer))
+        .over(order_by=history.c.timestamp)
+        .label("segment_id"),
+    ).subquery()
+
+    # Add column with rolling mean of the means (only inside of the segment)
+    history = Session.query(
+        history,
+        func.avg(history.c.mean)
```

Comment:
This `history.c.mean` here is the multisample mean which _might_ be for a single iteration (`n=1`), right?

Reply:
Yes, that's correct. This is how we've always done this, but I'm open to switching it up.